components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -54,124 +69,312 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
 }
 
 - (void)commonBannerViewInit {
-  self.backgroundColor = [UIColor whiteColor];
-
-  _buttons = [[NSArray alloc] init];
-  self.numberOfButtons = 1;
+  self.backgroundColor = UIColor.whiteColor;
 
+  // Create textLabel
   UILabel *textLabel = [[UILabel alloc] init];
-  textLabel.font = [UIFont systemFontOfSize:kTextFontSize];
-  textLabel.textColor = [UIColor colorWithWhite:0 alpha:kTextColorOpacity];
+  textLabel.translatesAutoresizingMaskIntoConstraints = NO;
+  textLabel.font = [MDCTypography body2Font];
+  textLabel.textColor = UIColor.blackColor;
+  textLabel.alpha = [MDCTypography body2FontOpacity];
   textLabel.numberOfLines = kTextNumberOfLineLimit;
   textLabel.lineBreakMode = NSLineBreakByTruncatingTail;
-  _textLabel = textLabel;
   [self addSubview:textLabel];
+  _textLabel = textLabel;
+
+  // Create imageView
+  UIImageView *imageView = [[UIImageView alloc] init];
+  imageView.translatesAutoresizingMaskIntoConstraints = NO;
+  if (@available(iOS 9.0, *)) {
+    [imageView.widthAnchor constraintEqualToConstant:kImageViewSideLength].active = YES;
+    [imageView.heightAnchor constraintEqualToConstant:kImageViewSideLength].active = YES;
+  }
+  imageView.contentMode = UIViewContentModeCenter;
+  imageView.clipsToBounds = YES;
+  [self addSubview:imageView];
+  _imageView = imageView;
+
+  // Create a button container to organize buttons
+  UIView *buttonContainerView = [[UIView alloc] init];
+  buttonContainerView.translatesAutoresizingMaskIntoConstraints = NO;
+  [self addSubview:buttonContainerView];
+  self.buttonContainerView = buttonContainerView;
+
+  // Create leadingButton and trailingButton
+  MDCButton *leadingButton = [[MDCButton alloc] init];
+  leadingButton.translatesAutoresizingMaskIntoConstraints = NO;
+  leadingButton.backgroundColor = UIColor.whiteColor;
+  [buttonContainerView addSubview:leadingButton];
+  _leadingButton = leadingButton;
+
+  MDCButton *trailingButton = [[MDCButton alloc] init];
+  trailingButton.translatesAutoresizingMaskIntoConstraints = NO;
+  trailingButton.backgroundColor = UIColor.whiteColor;
+  [buttonContainerView addSubview:trailingButton];
+  _trailingButton = trailingButton;
 }
 
 #pragma mark - Property Setter and Getter
 
-- (void)setText:(NSString *)text {
-  self.textLabel.text = text;
+- (BOOL)isImageSet {
```

Comment:
rename function: hasImage

it is more idiomatic to objective c and UIKit

Reply:
Addressed in 070341d.